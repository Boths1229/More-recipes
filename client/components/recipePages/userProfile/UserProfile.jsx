import React from 'react';
import avatar from '../../../public/images/avatar.png';

const UserProfile = (props) => {
  const { email, username } = props.userData;
  return (
    <div className="row mt-5 pb-3">
      <div className="col-sm-12 profile-bg user-profile">
        <img
          className="img-fluid rounded-circle"
          src={avatar}
          alt="avatar"
        />
        <h3>{username}</h3>
        <p>recipe added: 20
          <span className="vl ml-2 user-text" />
          <span>email: {email}</span>
          <span className="vl ml-2 user-text">favorite recipe: 20 </span>
        </p>
      </div>
    </div>
  );
};

export default UserProfile;
