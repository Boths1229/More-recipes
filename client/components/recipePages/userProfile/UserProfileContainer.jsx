import React, { Component } from 'react';
import miniToastr from 'mini-toastr';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import PropTypes, { any } from 'prop-types';
import ReactPaginate from 'react-paginate';
import { Lines } from 'react-preloading-component';
import getFavoriteRecipeAction from '../../../actionController/getFavoriteRecipe';
import getUserRecipeAction from '../../../actionController/getUserRecipe';
import Header from '../../common/Header';
import Footer from '../../common/Footer';
import FavoriteRecipeTitle from '../../../components/Include/FavoriteRecipeTitle';
import UserProfile from './UserProfile';
import RecipeGrid from '../../recipePages/RecipeGrid';
import upvoteRecipeAction from '../../../actionController/upvoteRecipe';
import downvoteRecipeAction from '../../../actionController/downvoteRecipe';
import favoriteRecipeAction from '../../../actionController/favoriteRecipe';


/**
 * @class UserProfileContainer
 *
 * @description container displaying user profile and favorited recipe of a user
 */
class UserProfileContainer extends Component {
  /**
   * @description create an instance of UserProfileContainer
   *
   * @param {props} props
   */
  constructor(props) {
    super(props);
    this.upvoteRecipe = this.upvoteRecipe.bind(this);
    this.downvoteRecipe = this.downvoteRecipe.bind(this);
    this.favoriteRecipe = this.favoriteRecipe.bind(this);
    this.handlePaginationChange = this.handlePaginationChange.bind(this);
  }
  /**
   * @description call the action to display user favorite recipe
   *
   * @param {props} props
   *
   * @return {undefined} call getRecipe
   */
  componentDidMount() {
    const { userId, page } = this.props;
    this.props.getUserRecipeAction(userId, page);    
    this.props.getFavoriteRecipeAction(userId, page);
  }
  /**
   * @description upvote a recipe
   *
   * @param {id} id id of recipe to be upvoted
   *
   * @return {void} calls upvoteRecipeAction
   */
  upvoteRecipe(id) {
    this.props.upvoteRecipeAction(id);
  }

  /**
   * @description downvote a recipe
   *
   * @param {id} id id of recipe to be updated
   *
   * @return {undefined} calls downvoteRecipeAction
   */
  downvoteRecipe(id) {
    if (this.props.authenticated) {
      return this.props.downvoteRecipeAction(id);
    }
    miniToastr.init();
    return miniToastr.error('Login to continue');
  }

  /**
   * @description favorite a recipe
   *
   * @param {id} id - id of recipe to be favorited
   *
   * @return {undefined} calls favoriteRecipeAction
   */
  favoriteRecipe(id) {
    this.props.favoriteRecipeAction(id);
  }

  handlePaginationChange(favoriteRecipes) {
    const { userId } = this.props;
    this.props.getFavoriteRecipeAction(userId, favoriteRecipes.selected);
  }


  /**
   * @description render - display component
   *
   * @return {JSX} return JSX
   */
  render() {
    return (
      <div>
        <Header />
        <div className="container">
          <UserProfile
            userData={this.props.userData}
            favoriteCount={this.props.favoriteCount}
            recipeCount={this.props.recipeCount}
          />
          <FavoriteRecipeTitle />
          <section className="row">
            {
             this.props.favoriteRecipes.isFetched ?
               <RecipeGrid
                 recipeData={this.props.favoriteRecipes.recipeData.data}
                 upvoteRecipe={this.upvoteRecipe}
                 downvoteRecipe={this.downvoteRecipe}
                 favoriteRecipe={this.favoriteRecipe}
               />
            :
               <Lines />
        }
          </section>
        </div>
        <ReactPaginate
          previousLabel="Previous"
          nextLabel="Next"
          breakLabel={<a href="">...</a>}
          breakClassName="page-link"
          onPageChange={this.handlePaginationChange}
          pageCount={this.props.page}
          containerClassName="pagination justify-content-center"
          pageLinkClassName="page-link"
          nextLinkClassName="page-link"
          previousLinkClassName="page-link"
          disabledClassName="disabled"
          pageClassName="page-item"
          previousClassName="page-item"
          nextClassName="page-item"
          activeClassName="active"
          subContainerClassName="pages pagination"
        />
        <Footer />
      </div>
    );
  }
}

UserProfileContainer.propTypes = {
  upvoteRecipeAction: PropTypes.func.isRequired,
  downvoteRecipeAction: PropTypes.func.isRequired,
  favoriteRecipeAction: PropTypes.func.isRequired,
  getFavoriteRecipeAction: PropTypes.func.isRequired,
  favoriteRecipes: PropTypes.objectOf(any).isRequired,
  authenticated: PropTypes.bool.isRequired,
  userData: PropTypes.objectOf(any).isRequired,
  userId: PropTypes.number.isRequired,
  page: PropTypes.number.isRequired
};

const mapStateToProps = state => ({
  authenticated: state.authReducer.isAuthenticated,
  recipes: state.recipeReducer,
  userId: state.authReducer.userData.id,
  page: state.getFavoriteRecipeReducer.page,
  userData: state.authReducer.userData,
  favoriteRecipes: state.getFavoriteRecipeReducer,
  favoriteCount: state.getFavoriteRecipeReducer.count,
  recipeCount: state.getUserRecipeReducer.count
});
/**
 * @description make actions available to AllRecipes as props
 *
 * @param {dispatch} dispatch
 *
 * @returns {undefined} call bindActionCreators function
 */
const mapDispatchToProps = dispatch => (
  bindActionCreators({
    upvoteRecipeAction,
    downvoteRecipeAction,
    favoriteRecipeAction,
    getFavoriteRecipeAction,
    getUserRecipeAction
  }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(UserProfileContainer);
