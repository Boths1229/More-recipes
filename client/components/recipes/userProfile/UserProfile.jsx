import React from 'react';

const UserProfile = props => (
  <div className="row pb-3">
    <div className="col-sm-12 profile-bg user-profile">
      <img className="img-fluid rounded-circle" src="../image/avatar.png" alt="avatar" />
      <h3>Sannikay</h3>
      <p>recipe added: 20
        <span className="vl user-text" />
        <span>email: kensanni@gmail.com </span>
        <span className="vl user-text">favorite recipe: 20</span>
      </p>
    </div>
  </div>
);

export default UserProfile;
