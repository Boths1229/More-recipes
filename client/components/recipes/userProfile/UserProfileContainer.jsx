import React, { Component } from 'react';
import miniToastr from 'mini-toastr';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import getFavoriteRecipeAction from '../../../actionController/getFavoriteRecipe';
import Header from '../../common/Header';
import Footer from '../../common/Footer';
import RecipeGrid from '../../recipes/RecipeGrid';
import upvoteRecipeAction from '../../../actionController/upvoteRecipe';
import downvoteRecipeAction from '../../../actionController/downvoteRecipe';
import favoriteRecipeAction from '../../../actionController/favoriteRecipe';


/**
 * @class UserProfileContainer
 *
 * @description container display user profile and favorited recipe of a user
 */
class UserProfileContainer extends Component {
  /**
   * @description call the action to display user favorite recipe
   *
   * @param {props} props
   *
   * @return {undefined} call getRecipe
   */
  componentDidMount() {
    const { userId } = this.props;
    console.log(userId);
    this.props.getFavoriteRecipeAction(userId);
  }
  /**
   * @description upvote a recipe
   *
   * @param {id} id id of recipe to be upvoted
   *
   * @return {void} calls upvoteRecipeAction
   */
  upvoteRecipe(id) {
    this.props.upvoteRecipeAction(id);
  }

  /**
   * @description downvote a recipe
   *
   * @param {id} id id of recipe to be updated
   *
   * @return {undefined} calls downvoteRecipeAction
   */
  downvoteRecipe(id) {
    if (this.props.authenticated) {
      return this.props.downvoteRecipeAction(id);
    }
    miniToastr.init();
    return miniToastr.error('Login to continue');
  }

  /**
   * @description favorite a recipe
   *
   * @param {id} id - id of recipe to be favorited
   *
   * @return {undefined} calls favoriteRecipeAction
   */
  favoriteRecipe(id) {
    this.props.favoriteRecipeAction(id);
  }
  /**
   * @description render - display component
   *
   * @return {JSX} return JSX
   */
  render() {
    console.log('***', this.props.favoriteRecipes.recipeData.data);
    return (
      <div>
        <Header />
        {
          this.props.favoriteRecipes.isFetched ?
            <RecipeGrid
              recipeData={this.props.favoriteRecipes.recipeData.data}
              upvoteRecipe={this.upvoteRecipe}
              downvoteRecipe={this.downvoteRecipe}
              favoriteRecipe={this.favoriteRecipe}
            />
          :
            <h1 style={{marginTop: '150px'}}>I am Loading</h1>
        }
        <Footer />
      </div>
    );
  }
}
const mapStateToProps = state => (
  console.log('auth', state.getFavoriteRecipeReducer[0]),
  {
    authenticated: state.authReducer.isAuthenticated,
    recipes: state.recipeReducer,
    userId: state.authReducer.userData.id,
    favoriteRecipes: state.getFavoriteRecipeReducer[0]
  });
/**
 * @description make actions available to AllRecipes as props
 *
 * @param {dispatch} dispatch
 *
 * @returns {undefined} call bindActionCreators function
 */
const mapDispatchToProps = dispatch => (
  bindActionCreators({
    upvoteRecipeAction,
    downvoteRecipeAction,
    favoriteRecipeAction,
    getFavoriteRecipeAction
  }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(UserProfileContainer);
