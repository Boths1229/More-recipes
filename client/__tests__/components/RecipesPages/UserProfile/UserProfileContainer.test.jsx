import React from 'react';
import { shallow } from 'enzyme';
import { UserProfileContainer } from '../../../../components/recipePages/userProfile/UserProfileContainer';

describe('user profile container component', () => {
  it('matches snapshot', () => {
    const props = {
      favoriteRecipes: {
        recipeData: []
      },
      history: {
        push: jest.fn()
      }
    };
    const output = shallow(<UserProfileContainer {...props} />);
    expect(output).toMatchSnapshot();
  });
});
