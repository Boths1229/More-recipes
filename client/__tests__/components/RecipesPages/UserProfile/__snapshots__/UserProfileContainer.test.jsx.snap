// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user profile container component matches snapshot 1`] = `
<div>
  <Connect(Header) />
  <div
    className="container"
  >
    <UserProfile />
    <FavoriteRecipeTitle />
    <section>
      <n
        color="#FF4601"
        size={30}
      />
    </section>
  </div>
  <PaginationBoxView
    activeClassName="active"
    breakClassName="page-link"
    breakLabel={
      <a
        href=""
      >
        ...
      </a>
    }
    containerClassName="pagination justify-content-center"
    disableInitialCallback={false}
    disabledClassName="disabled"
    marginPagesDisplayed={3}
    nextClassName="page-item"
    nextLabel="Next"
    nextLinkClassName="page-link"
    onPageChange={[Function]}
    pageClassName="page-item"
    pageCount={10}
    pageLinkClassName="page-link"
    pageRangeDisplayed={2}
    previousClassName="page-item"
    previousLabel="Previous"
    previousLinkClassName="page-link"
    subContainerClassName="pages pagination"
  />
  <Footer />
</div>
`;
